src/components/DropMenu/DropMenu.jsx

```diff
@@ -0,0 +1,467 @@
+		isDisabled: bool,
+		/**
+		 * Renders the flyout menu adjacent to the control.
+		 */
+		isExpanded: bool,
+		/**
+		 * Sets the direction the flyout menu will render relative to the control.
+		 */
+		direction: oneOf(['down', 'up']),
+		/**
+		 * An array of currently selected `DropMenu.Option` indices.
+		 */
+		selectedIndices: arrayOf(number),
+		/**
+		 * The currently focused index of `DropMenu.Option`. Can also be `null`.
+		 */
+		focusedIndex: number,
+		/**
+		 * The `id` of the flyout menu portal element that is appended to `document.body`. Defaults to a generated id.
+		 */
+		portalId: string,
+		/**
+		 * Called when collapsed and the control is clicked, or when the control has focus and the Down Arrow or Space key is pressed.
+		 */
+		onExpand: func,
+		/**
+		 * Called when expanded and the user clicks the control or outside of the menu, or when the control has focus and the Escape key is pressed.
+		 */
+		onCollapse: func,
+		/**
+		 * Called when an option is clicked, or when an option has focus and the Space or Enter key is pressed.
+		 * Has the signature `(optionIndex) => {}` where optionIndex can be a number or `null`.
+		 */
+		onSelect: func,
+		/**
+		 * Called when expanded and the the Down Arrow key is pressed. Not called when focus is on the last option.
+		 */
+		onFocusNext: func,
+		/**
+		 * Called when expanded and the the Up Arrow key is pressed. Not called when focus is on the first option.
+		 */
+		onFocusPrev: func,
+		/**
+		 * Called when `direction` is `down` and the bottom of the flyout menu is clipping the screen boundary.
+		 */
+		onBelowFold: func,
+		/**
+		 * Called when `direction` is `up` and there is enough space below the control to render the flyout menu without clipping the screen boundary.
+		 */
+		onAboveFold: func,
+		/**
+		 * Called when the mouse moves over an option.
+		 * Has the signature `(optionIndex) => {}` where optionIndex can be a number or `null`.
+		 */
+		onFocusOption: func,
+		/**
+		 * *Child Element* - The control target which the flyout menu is anchored to. Only one `Control` is used.
+		 */
+		Control: any,
+		/**
+		 * *Child Element* - These are menu options. The `optionIndex` is in-order of rendering regardless of group nesting, starting with index `0`.
+		 * Each `Option` may be passed a prop called `isDisabled` to disable selection of that `Option`.
+		 */
+		Option: any,
+		/**
+		 * *Child Element* - Used to group `Option`s within the menu. Any non-`Option`s passed in will be rendered as a label for the group.
+		 */
+		OptionGroup: any,
+		/**
+		 * *Child Element* - A special kind of `Option` that is always rendered at the top of the menu and has an `optionIndex` of `null`. Useful for unselect.
+		 */
+		NullOption: any
+	},
+
+	getDefaultProps() {
+		return {
+			isDisabled: false,
+			isExpanded: false,
+			direction: 'down',
+			selectedIndices: [],
+			focusedIndex: null,
+			portalId: 'DropMenu-Portal-' + Math.random().toString(16).substr(2),
```

Comment:
We _really_ should abstract this out to a utility function since it's being used in a number of places in the library. :)